var/spack/repos/builtin/packages/nvpl-fft/package.py

```diff
@@ -36,3 +38,20 @@ def url_for_version(self, version):
 
     def install(self, spec, prefix):
         install_tree(".", prefix)
+
+    @property
+    def headers(self):
+        return find_all_headers(self.spec.prefix.include)
+
+    @property
+    def libs(self):
+        return find_libraries("libnvpl_fftw", self.spec.prefix.lib, shared=True, recursive=True)
+
+    @run_after("install")
+    def fix_include(self):
+        subdir = os.path.join(self.prefix.include, "nvpl_fftw")  # include/nvpl_fftw/
+        for file in os.listdir(subdir):
+            file_symlink = os.path.join(self.prefix.include, os.path.basename(file))
+            # nvpl_fft_version.h is duplicated in include/ and include/nvpl_fftw/
+            if not os.path.exists(file_symlink):
+                symlink(os.path.join(subdir, file), file_symlink)
```

Comment:
I don't know if this is the best approach (probably not...), but `intel-onaapi-mkl` does something similar:
https://github.com/spack/spack/blob/7d79648cb507865309ab2c8f6e54743377e645c3/var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py#L302-L311